src/python/foglamp/core/statistics_db_services.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+"""Storage Services as needed by Processes
+"""
+
+import asyncpg
+from collections import OrderedDict
+
+__author__ = "Amarendra Kumar Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+__DB_NAME = 'foglamp'
+
+
+async def read_statistics():
+    conn = await asyncpg.connect(database=__DB_NAME)
+    query = """
+        SELECT key, description, value, previous_value, ts FROM statistics ORDER BY key
+    """
+
+    stmt = await conn.prepare(query)
+
+    rows = await stmt.fetch()
+
+    columns = ('key',
+               'description',
+               'value',
+               'previous_value',
+               'ts'
+               )
+
+    results = {}
+    for row in rows:
+        temp = OrderedDict(zip(columns, row))
+        results.update({temp['key']: temp['value']})
+
+    await conn.close()
+
+    return results
+
+async def read_statistics_history(limit=None):
+    conn = await asyncpg.connect(database=__DB_NAME)
+    _limit_clause = " LIMIT $1" if limit else " "
+
+    query = """
+                SELECT date_trunc('second', history_ts::timestamptz)::varchar as history_ts,
+                        key,
+                        value FROM statistics_history
+                WHERE history_ts IN (SELECT distinct history_ts FROM statistics_history ORDER BY history_ts DESC {limit_clause})
```

Comment:
This only works if history_ts is the same for all statistics in a single grouping... which it is not currently.